### PR TITLE
Removes coalign by solar rotation completely

### PR DESCRIPTION
**Pull request by Deus1704**
## PR Description
#203 
Following our discussion on the removal of the methods `calculate_solar_rotate_shift` and `mapsequence_coalign_by_rotation`, this PR removes them along with removal of the example gallery and the unit tests.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by wtbarnes** [comment]
The doc build fail is real. It looks like there are still a few references to the removed functions somewhere in the docs. I'm not sure what's going on with the online build.

File: changelog/206.removal.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The coalignment methods ``calculate_solar_rotate_shift`` and ``mapsequence_coalign_by_rotation`` present in the `sunkit_image.coalignment` have been removed, with no replacement.
+Correspondingly the gallery example ``Compensating for Solar Rotation`` making use of these methods are removed.
+This is because they are worse versions of the functionality already present in sunpy to reproject images given some model of differential rotation.
```
Comment (wtbarnes) [inline changelog/206.removal.rst:3]: This is not necessary.

```suggestion
```

**Review by wtbarnes — CHANGES_REQUESTED** [review]
As noted above, the doc build failure still needs fixing.

**Comment by Deus1704** [comment]
> As noted above, the doc build failure still needs fixing.

Yeah there were few references in the legacy changelog dating back to the initial addition of these functions.

**Comment by nabobalis** [comment]
Online test is a file download failure.

**Comment by Deus1704** [comment]
> Online test is a file download failure.

Yeah, but I believe we should look into it? because the same issue is occurring when I run the online test suite locally and the file being downloaded is a sunpy hosted sample data, so ideally this failure shouldn't occur?

**Comment by nabobalis** [comment]
> > Online test is a file download failure.
> 
> Yeah, but I believe we should look into it? because the same issue is occurring when I run the online test suite locally and the file being downloaded is a sunpy hosted sample data, so ideally this failure shouldn't occur?

Yeah but it isnt due to this PR so we can gloss over it for this.

**Pull request merged**